src/main/java/com/google/api/generator/spring/composer/SpringAutoConfigClassComposer.java

```diff
@@ -689,77 +620,107 @@ private static MethodDefinition createSettingsBeanMethod(
-              String propertyName =
-                  CaseFormat.LOWER_CAMEL.to(
-                      CaseFormat.UPPER_CAMEL, Joiner.on("").join(methodAndPropertyName));
-
-              TypeNode propertyType = defaultVal.type();
-              if (propertyType.equals(TypeNode.DOUBLE)) {
-                propertyType = TypeNode.DOUBLE_OBJECT;
-              }
-              MethodInvocationExpr currentRetrySettingProperty =
-                  MethodInvocationExpr.builder()
-                      .setMethodName(String.format("get%s", propertyName))
-                      .setReturnType(propertyType)
-                      .setExprReferenceExpr(thisClientPropertiesVarExpr)
-                      .build();
-              RelationalOperationExpr currentRetrySettingPropertyIsNull =
-                  RelationalOperationExpr.notEqualToWithExprs(
-                      currentRetrySettingProperty, ValueExpr.createNullExpr());
-
-              // annotateTextRetrySettingsBuilder.setInitialRetryDelay(this.clientProperties.getAnnotateTextInitialRetryDelay());
-              Variable retrySettingBuilderVariable =
-                  Variable.builder()
-                      .setName(
-                          String.format("%sRetrySettingBuilder", methodName)) // extract method name
-                      .setType(STATIC_TYPES.get("RetrySettings.Builder"))
-                      .build();
-              MethodInvocationExpr retrySettingsBuilderChain =
-                  MethodInvocationExpr.builder()
-                      .setExprReferenceExpr(VariableExpr.withVariable(retrySettingBuilderVariable))
-                      .setMethodName(String.format("set%s", settingName))
-                      .setArguments(currentRetrySettingProperty)
-                      .build();
-
-              IfStatement currentRetrySettingPropertyIfStatement =
-                  createIfStatement(
-                      currentRetrySettingPropertyIsNull,
-                      Arrays.asList(
-                          ExprStatement.withExpr(retrySettingsBuilderChain),
-                          LoggerUtils.createLoggerStatement(
-                              LoggerUtils.concatManyWithExprs(
-                                  ValueExpr.withValue(
-                                      StringObjectValue.withValue(propertyName + " set to ")),
-                                  currentRetrySettingProperty),
-                              types)),
-                      null);
-              statements.add(currentRetrySettingPropertyIfStatement);
-              return statements;
-            },
-            (String methodName) -> {
-              List<Statement> statements = new ArrayList<>();
-              statements.add(setRetrySettingsForMethod(methodName, settingBuilderVariable));
-              return statements;
-            });
-    bodyStatements.addAll(retrySettings);
+
+    // If service-level properties configured, update retry settings for each method
```

Comment:
Perhaps add description of logic between setting service-level vs method-level retries to the javadoc comment of this `EchoSettings` bean?

Reply:
Makes sense, updated! 

(FYI @diegomarquezp I only changed the part relevant to retry settings in https://github.com/googleapis/gapic-generator-java/pull/1143/commits/4d55a7a640248eeebf61cbde3c1300bf54d0d9c9 , hopefully it won't conflict too much with your updates for `ServiceClient` => `ServiceSettings` bean)